### ISSUE TITLE
Tab "Audit Vulnerabilities" in project view is empty

### ISSUE THREAD

**Issue by a5a351e7** [DependencyTrack/dependency-track#4300]
### Current Behavior

In an existing project, the “Audit Vulnerabilities” tab does not show any vulnerabilities, although these should be displayed.

![Image](https://github.com/user-attachments/assets/2f023444-5536-463a-8b1b-0e5981548df2)

In the network console of the browser, the following two requests could not be executed successfully:

![Image](https://github.com/user-attachments/assets/6e229e4f-c077-49cd-92cc-a241354dc6b7)

The following server is logged on the server:

```
dtrack-apiserver-1  | 2024-10-24 07:56:04,731 ERROR [GlobalExceptionHandler] Uncaught internal server error [requestId=4353b7e8-5d62-406a-96e8-f7677eff5ca8]
dtrack-apiserver-1  | java.lang.ClassCastException: class java.lang.Long cannot be cast to class java.lang.Integer (java.lang.Long and java.lang.Integer are in module java.base of loader 'bootstrap')
dtrack-apiserver-1  |   at org.dependencytrack.persistence.VulnerabilityQueryManager.lambda$getVulnerabilityAliases$3(VulnerabilityQueryManager.java:729)
dtrack-apiserver-1  |   at java.base/java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
dtrack-apiserver-1  |   at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(Unknown Source)
dtrack-apiserver-1  |   at java.base/java.util.stream.AbstractPipeline.copyInto(Unknown Source)
dtrack-apiserver-1  |   at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
dtrack-apiserver-1  |   at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(Unknown Source)
dtrack-apiserver-1  |   at java.base/java.util.stream.AbstractPipeline.evaluate(Unknown Source)
dtrack-apiserver-1  |   at java.base/java.util.stream.ReferencePipeline.collect(Unknown Source)
dtrack-apiserver-1  |   at org.dependencytrack.persistence.VulnerabilityQueryManager.getVulnerabilityAliases(VulnerabilityQueryManager.java:740)
dtrack-apiserver-1  |   at org.dependencytrack.persistence.QueryManager.getVulnerabilityAliases(QueryManager.java:1128)
dtrack-apiserver-1  |   at org.dependencytrack.persistence.FindingsQueryManager.getFindings(FindingsQueryManager.java:296)
dtrack-apiserver-1  |   at org.dependencytrack.persistence.QueryManager.getFindings(QueryManager.java:1162)
dtrack-apiserver-1  |   at org.dependencytrack.resources.v1.FindingResource.getFindingsByProject(FindingResource.java:125)
dtrack-apiserver-1  |   at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source)
dtrack-apiserver-1  |   at java.base/java.lang.reflect.Method.invoke(Unknown Source)
dtrack-apiserver-1  |   at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory.lambda$static$0(ResourceMethodInvocationHandlerFactory.java:52)
dtrack-apiserver-1  |   at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:146)
dtrack-apiserver-1  |   at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:189)
dtrack-apiserver-1  |   at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$ResponseOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:176)
dtrack-apiserver-1  |   at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:93)
dtrack-apiserver-1  |   at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:478)
dtrack-apiserver-1  |   at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:400)
dtrack-apiserver-1  |   at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:81)
dtrack-apiserver-1  |   at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:274)
dtrack-apiserver-1  |   at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248)
dtrack-apiserver-1  |   at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244)
dtrack-apiserver-1  |   at org.glassfish.jersey.internal.Errors.process(Errors.java:292)
dtrack-apiserver-1  |   at org.glassfish.jersey.internal.Errors.process(Errors.java:274)
dtrack-apiserver-1  |   at org.glassfish.jersey.internal.Errors.process(Errors.java:244)
dtrack-apiserver-1  |   at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:266)
dtrack-apiserver-1  |   at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:253)
dtrack-apiserver-1  |   at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:696)
dtrack-apiserver-1  |   at org.glassfish.jersey.servlet.WebComponent.serviceImpl(WebComponent.java:397)
dtrack-apiserver-1  |   at org.glassfish.jersey.servlet.WebComponent.service(WebComponent.java:349)
dtrack-apiserver-1  |   at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:358)
dtrack-apiserver-1  |   at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:312)
dtrack-apiserver-1  |   at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:205)
dtrack-apiserver-1  |   at org.eclipse.jetty.ee10.servlet.ServletHolder$NotAsync.service(ServletHolder.java:1379)
dtrack-apiserver-1  |   at org.eclipse.jetty.ee10.servlet.ServletHolder.handle(ServletHolder.java:736)
dtrack-apiserver-1  |   at org.eclipse.jetty.ee10.servlet.ServletHandler$ChainEnd.doFilter(ServletHandler.java:1614)
dtrack-apiserver-1  |   at alpine.server.filters.ContentSecurityPolicyFilter.doFilter(ContentSecurityPolicyFilter.java:225)
dtrack-apiserver-1  |   at org.eclipse.jetty.ee10.servlet.FilterHolder.doFilter(FilterHolder.java:205)
dtrack-apiserver-1  |   at org.eclipse.jetty.ee10.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1586)
dtrack-apiserver-1  |   at alpine.server.filters.ClickjackingFilter.doFilter(ClickjackingFilter.java:93)
dtrack-apiserver-1  |   at org.eclipse.jetty.ee10.servlet.FilterHolder.doFilter(FilterHolder.java:205)
dtrack-apiserver-1  |   at org.eclipse.jetty.ee10.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1586)
dtrack-apiserver-1  |   at alpine.server.filters.WhitelistUrlFilter.doFilter(WhitelistUrlFilter.java:166)
dtrack-apiserver-1  |   at org.eclipse.jetty.ee10.servlet.FilterHolder.doFilter(FilterHolder.java:208)
dtrack-apiserver-1  |   at org.eclipse.jetty.ee10.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1586)
dtrack-apiserver-1  |   at org.eclipse.jetty.ee10.servlet.ServletHandler$MappedServlet.handle(ServletHandler.java:1547)
dtrack-apiserver-1  |   at org.eclipse.jetty.ee10.servlet.ServletChannel.dispatch(ServletChannel.java:824)
dtrack-apiserver-1  |   at org.eclipse.jetty.ee10.servlet.ServletChannel.handle(ServletChannel.java:436)
dtrack-apiserver-1  |   at org.eclipse.jetty.ee10.servlet.ServletHandler.handle(ServletHandler.java:464)
dtrack-apiserver-1  |   at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:575)
dtrack-apiserver-1  |   at org.eclipse.jetty.ee10.servlet.SessionHandler.handle(SessionHandler.java:703)
dtrack-apiserver-1  |   at org.eclipse.jetty.server.handler.ContextHandler.handle(ContextHandler.java:1060)
dtrack-apiserver-1  |   at org.eclipse.jetty.server.Server.handle(Server.java:181)
dtrack-apiserver-1  |   at org.eclipse.jetty.server.internal.HttpChannelState$HandlerInvoker.run(HttpChannelState.java:661)
dtrack-apiserver-1  |   at org.eclipse.jetty.server.internal.HttpConnection.onFillable(HttpConnection.java:406)
dtrack-apiserver-1  |   at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:322)
dtrack-apiserver-1  |   at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:99)
dtrack-apiserver-1  |   at org.eclipse.jetty.io.SelectableChannelEndPoint$1.run(SelectableChannelEndPoint.java:53)
dtrack-apiserver-1  |   at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.runTask(AdaptiveExecutionStrategy.java:478)
dtrack-apiserver-1  |   at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.consumeTask(AdaptiveExecutionStrategy.java:441)
dtrack-apiserver-1  |   at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.tryProduce(AdaptiveExecutionStrategy.java:293)
dtrack-apiserver-1  |   at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.run(AdaptiveExecutionStrategy.java:201)
dtrack-apiserver-1  |   at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:311)
dtrack-apiserver-1  |   at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:979)
dtrack-apiserver-1  |   at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.doRunJob(QueuedThreadPool.java:1209)
dtrack-apiserver-1  |   at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1164)
dtrack-apiserver-1  |   at java.base/java.lang.Thread.run(Unknown Source)
dtrack-apiserver-1  | 2024-10-24 07:56:04,735 ERROR [GlobalExceptionHandler] Uncaught internal server error [requestId=53828d37-1f8f-486a-8c9e-d79179a88077]
dtrack-apiserver-1  | java.lang.ClassCastException: class java.lang.Long cannot be cast to class java.lang.Integer (java.lang.Long and java.lang.Integer are in module java.base of loader 'bootstrap')
dtrack-apiserver-1  |   at org.dependencytrack.persistence.VulnerabilityQueryManager.lambda$getVulnerabilityAliases$3(VulnerabilityQueryManager.java:729)
dtrack-apiserver-1  |   at java.base/java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
dtrack-apiserver-1  |   at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(Unknown Source)
dtrack-apiserver-1  |   at java.base/java.util.stream.AbstractPipeline.copyInto(Unknown Source)
dtrack-apiserver-1  |   at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
dtrack-apiserver-1  |   at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(Unknown Source)
dtrack-apiserver-1  |   at java.base/java.util.stream.AbstractPipeline.evaluate(Unknown Source)
dtrack-apiserver-1  |   at java.base/java.util.stream.ReferencePipeline.collect(Unknown Source)
dtrack-apiserver-1  |   at org.dependencytrack.persistence.VulnerabilityQueryManager.getVulnerabilityAliases(VulnerabilityQueryManager.java:740)
dtrack-apiserver-1  |   at org.dependencytrack.persistence.QueryManager.getVulnerabilityAliases(QueryManager.java:1128)
dtrack-apiserver-1  |   at org.dependencytrack.persistence.FindingsQueryManager.getFindings(FindingsQueryManager.java:296)
dtrack-apiserver-1  |   at org.dependencytrack.persistence.QueryManager.getFindings(QueryManager.java:1162)
dtrack-apiserver-1  |   at org.dependencytrack.resources.v1.FindingResource.getFindingsByProject(FindingResource.java:125)
dtrack-apiserver-1  |   at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source)
dtrack-apiserver-1  |   at java.base/java.lang.reflect.Method.invoke(Unknown Source)
dtrack-apiserver-1  |   at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory.lambda$static$0(ResourceMethodInvocationHandlerFactory.java:52)
dtrack-apiserver-1  |   at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:146)
dtrack-apiserver-1  |   at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:189)
dtrack-apiserver-1  |   at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$ResponseOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:176)
dtrack-apiserver-1  |   at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:93)
dtrack-apiserver-1  |   at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:478)
dtrack-apiserver-1  |   at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:400)
dtrack-apiserver-1  |   at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:81)
dtrack-apiserver-1  |   at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:274)
dtrack-apiserver-1  |   at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248)
dtrack-apiserver-1  |   at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244)
dtrack-apiserver-1  |   at org.glassfish.jersey.internal.Errors.process(Errors.java:292)
dtrack-apiserver-1  |   at org.glassfish.jersey.internal.Errors.process(Errors.java:274)
dtrack-apiserver-1  |   at org.glassfish.jersey.internal.Errors.process(Errors.java:244)
dtrack-apiserver-1  |   at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:266)
dtrack-apiserver-1  |   at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:253)
dtrack-apiserver-1  |   at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:696)
dtrack-apiserver-1  |   at org.glassfish.jersey.servlet.WebComponent.serviceImpl(WebComponent.java:397)
dtrack-apiserver-1  |   at org.glassfish.jersey.servlet.WebComponent.service(WebComponent.java:349)
dtrack-apiserver-1  |   at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:358)
dtrack-apiserver-1  |   at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:312)
dtrack-apiserver-1  |   at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:205)
dtrack-apiserver-1  |   at org.eclipse.jetty.ee10.servlet.ServletHolder$NotAsync.service(ServletHolder.java:1379)
dtrack-apiserver-1  |   at org.eclipse.jetty.ee10.servlet.ServletHolder.handle(ServletHolder.java:736)
dtrack-apiserver-1  |   at org.eclipse.jetty.ee10.servlet.ServletHandler$ChainEnd.doFilter(ServletHandler.java:1614)
dtrack-apiserver-1  |   at alpine.server.filters.ContentSecurityPolicyFilter.doFilter(ContentSecurityPolicyFilter.java:225)
dtrack-apiserver-1  |   at org.eclipse.jetty.ee10.servlet.FilterHolder.doFilter(FilterHolder.java:205)
dtrack-apiserver-1  |   at org.eclipse.jetty.ee10.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1586)
dtrack-apiserver-1  |   at alpine.server.filters.ClickjackingFilter.doFilter(ClickjackingFilter.java:93)
dtrack-apiserver-1  |   at org.eclipse.jetty.ee10.servlet.FilterHolder.doFilter(FilterHolder.java:205)
dtrack-apiserver-1  |   at org.eclipse.jetty.ee10.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1586)
dtrack-apiserver-1  |   at alpine.server.filters.WhitelistUrlFilter.doFilter(WhitelistUrlFilter.java:166)
dtrack-apiserver-1  |   at org.eclipse.jetty.ee10.servlet.FilterHolder.doFilter(FilterHolder.java:208)
dtrack-apiserver-1  |   at org.eclipse.jetty.ee10.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1586)
dtrack-apiserver-1  |   at org.eclipse.jetty.ee10.servlet.ServletHandler$MappedServlet.handle(ServletHandler.java:1547)
dtrack-apiserver-1  |   at org.eclipse.jetty.ee10.servlet.ServletChannel.dispatch(ServletChannel.java:824)
dtrack-apiserver-1  |   at org.eclipse.jetty.ee10.servlet.ServletChannel.handle(ServletChannel.java:436)
dtrack-apiserver-1  |   at org.eclipse.jetty.ee10.servlet.ServletHandler.handle(ServletHandler.java:464)
dtrack-apiserver-1  |   at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:575)
dtrack-apiserver-1  |   at org.eclipse.jetty.ee10.servlet.SessionHandler.handle(SessionHandler.java:703)
dtrack-apiserver-1  |   at org.eclipse.jetty.server.handler.ContextHandler.handle(ContextHandler.java:1060)
dtrack-apiserver-1  |   at org.eclipse.jetty.server.Server.handle(Server.java:181)
dtrack-apiserver-1  |   at org.eclipse.jetty.server.internal.HttpChannelState$HandlerInvoker.run(HttpChannelState.java:661)
dtrack-apiserver-1  |   at org.eclipse.jetty.server.internal.HttpConnection.onFillable(HttpConnection.java:406)
dtrack-apiserver-1  |   at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:322)
dtrack-apiserver-1  |   at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:99)
dtrack-apiserver-1  |   at org.eclipse.jetty.io.SelectableChannelEndPoint$1.run(SelectableChannelEndPoint.java:53)
dtrack-apiserver-1  |   at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.runTask(AdaptiveExecutionStrategy.java:478)
dtrack-apiserver-1  |   at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.consumeTask(AdaptiveExecutionStrategy.java:441)
dtrack-apiserver-1  |   at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.tryProduce(AdaptiveExecutionStrategy.java:293)
dtrack-apiserver-1  |   at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.run(AdaptiveExecutionStrategy.java:201)
dtrack-apiserver-1  |   at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:311)
dtrack-apiserver-1  |   at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:979)
dtrack-apiserver-1  |   at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.doRunJob(QueuedThreadPool.java:1209)
dtrack-apiserver-1  |   at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1164)
dtrack-apiserver-1  |   at java.base/java.lang.Thread.run(Unknown Source)
```

The database schema is:
```
column name	#	Data Type	Not Null	Auto Increment	Key	Default	Extra	Expression	Comment
ID	1	bigint(20)	true	true	PRI	[NULL]	auto_increment	[NULL]	
CVE_ID	2	varchar(255)	false	false	MUL	NULL		[NULL]	
GHSA_ID	3	varchar(255)	false	false	MUL	NULL		[NULL]	
GSD_ID	4	varchar(255)	false	false	MUL	NULL		[NULL]	
INTERNAL_ID	5	varchar(255)	false	false	MUL	NULL		[NULL]	
OSV_ID	6	varchar(255)	false	false	MUL	NULL		[NULL]	
SONATYPE_ID	7	varchar(255)	false	false	MUL	NULL		[NULL]	
UUID	8	varchar(36)	true	false	UNI	[NULL]		[NULL]	
VULNDB_ID	9	varchar(255)	false	false	MUL	NULL		[NULL]	
SNYK_ID	10	varchar(255)	false	false	MUL	NULL		[NULL]	
```

### Steps to Reproduce

1. Login to Dependency-Track
2. Open a project with vulnerabilities
3. Click on tab Audit Vulnerabilities

### Expected Behavior

The list of seven vulnerabilities should be shown.

### Dependency-Track Version

4.12.0

### Dependency-Track Distribution

Container Image

### Database Server

N/A

### Database Server Version

MariaDB 10.5

### Browser

Microsoft Edge

### Checklist

- [x] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/master/CONTRIBUTING.md#filing-issues)
- [x] I have checked the [existing issues](https://github.com/DependencyTrack/dependency-track/issues) for whether this defect was already reported

**Comment by a5a351e7** [comment]
It seems that an unauthorized type conversion takes place in the following [line](https://github.com/DependencyTrack/dependency-track/blob/2c0755b857dd604acdffda86566cbeaf3ac6b01a/src/main/java/org/dependencytrack/persistence/VulnerabilityQueryManager.java#L729). The database model also suggests (bigint(20)) a Long rather than an Integer.

**Comment by a5a351e7** [comment]
@Malaydewangan09 Sure! Do you got the same issue?

**Comment by Malaydewangan09** [comment]
@a5a351e7 I see that you got this issue in version 4.12.0
I tested this issue but I guess this is fixed for version 4.13.0

**Comment by a5a351e7** [comment]
@Malaydewangan09 Thank you! I decided to switch from MariaDB to PostgreSQL because of the non-recommended support. Therefore, the issue here has become obsolete.
